src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -65,7 +65,13 @@ def iter_path_entries():
                     if entry == "<PATH>":
                         path = self._options_env.get("PATH")
                         if path:
-                            yield from path.split(os.pathsep)
+                            for prefix in path.split(os.pathsep):
+                                if prefix.startswith("$") or prefix.startswith("~"):
+                                    logger.warning(
+                                        f"unescaped literal prefix {prefix} in PATH ignored. Check the value of the PATH variable in your SHELL"
```

Comment:
I reword to make it more descriptive. How about? `f"Ignored unexpanded path prefix `{prefix}` in the `PATH` environment variable. Please check the value of the `PATH` environment variable in your shell."`